sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/DateExpressionsSuite.scala

```diff
@@ -836,7 +836,7 @@ class DateExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     }
   }
 
-  test("datediff") {
+  test("datediff returns an integer") {
```

Comment:
We had better have `withSQLConf(SQLConf.ANSI_ENABLED.key -> "true", SQLConf.DIALECT.key -> "...` here too.

Reply:
In that case, it will not test `SQLConf.ANSI_ENABLED.key -> "false"` which is default behavior.

Reply:
Sorry, I blindly copied. What I mean was setting `SQLConf` with the default configuration. The default configurations can be changed easily.

Reply:
I think it would be better to add asserts to check this assumption otherwise if defaults for the SQL config will be changed in the future, this test will not check default behavior.